mantidproject/mantid
# Fixes whitespace and run loading bugs in Muon GUIs

**Description of work.**
Changes to the run utils to deal with errant whitespaces in runs (e.g. 60715- 7) and runs that straddle a decade but the second number was lower than the final digit of the first  (e.g. 60719-2). The changes mean that neither issue produces an error message or fails to load at all.

**To test:**
1. Open Muon Analysis 2 GUI
2. Make sure the instrument selected is MUSR
3. Load runs ```62259-2``` 
4. An error message will appear about field alignments but the runs will still load (previously nothing would load at all)
5. Change instrument to EMU
6. Load runs 20882- 3
7. Runs should load correctly
8. Open Frequency Domain Analysis GUI and repeat steps 2 to 7


Fixes #32004 and #32073

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
